chaudloader/src/hooks/stage1.rs

```diff
@@ -133,6 +138,37 @@ unsafe fn on_game_load(
     }
 }
 
+unsafe fn on_pck_load(
+    sound_engine_class: *mut u8,
+    pck_file_name: *const u16,
+    unk_pck_ptr: *mut u8
+) -> i32 {
+    let pck_wstr = std::ffi::OsString::from_wide(
+        std::slice::from_raw_parts(pck_file_name,
+            winapi::um::winbase::lstrlenW(pck_file_name) as usize
+        )
+    );
+    // Called first but actually called last?
+    let return_val = mmbnlc_PckLoad.call(sound_engine_class, pck_file_name, unk_pck_ptr);
+    let mod_audio = mods::MODAUDIOFILES.get().unwrap().lock().unwrap();
+    if let Some(pck_str) = pck_wstr.to_str() {
+        match pck_str {
+            "Vol1.pck" | "Vol2.pck" => {
+                for pck in &mod_audio.pcks {
+                    let mod_pck_wstr = pck
+                        .encode_wide()
+                        .chain(std::iter::once(0))
+                        .collect::<Vec<_>>();
+                    let mod_pck_wstr_ptr = mod_pck_wstr[..].as_ptr();
```

Comment:
I don't think you need the `[..]` to first create a slice here, as far as I can tell you can just call `Vec.as_ptr()` directly.